corehq/apps/api/es.py

```diff
@@ -145,6 +146,8 @@ def run_query(self, es_query, es_type=None):
 
         try:
             es_results = self.es.search(self.index, es_type, body=es_query)
+            if es_results.get('_shards', {}).get('failed'):
+                datadog_counter('commcare.es.partial_results', value=1)
```

Comment:
nit: these two lines could be pulled out int a function `report_any_shard_failure(es_search_result)`. (It would be sweet if we could somehow get the `search` method itself to log these so we knew all cases are covered, but I think we don't own that class so don't think that's practical.)